pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicsAuthTest.java

```diff
@@ -0,0 +1,236 @@
+
+public class TopicsAuthTest extends MockedPulsarServiceBaseTest {
+
+    private final String testLocalCluster = "test";
+    private final String testTenant = "my-tenant";
+    private final String testNamespace = "my-namespace";
+    private final String testTopicName = "my-topic";
+
+    private static final SecretKey SECRET_KEY = AuthTokenUtils.createSecretKey(SignatureAlgorithm.HS256);
+    private static final String ADMIN_TOKEN = Jwts.builder().setSubject("admin").signWith(SECRET_KEY).compact();
+    private static final String PRODUCE_TOKEN = Jwts.builder().setSubject("producer").signWith(SECRET_KEY).compact();
+    private static final String CONSUME_TOKEN = Jwts.builder().setSubject("consumer").signWith(SECRET_KEY).compact();
+
+    @Override
+    @BeforeMethod
+    protected void setup() throws Exception {
+        // enable auth&auth and use JWT at broker
+        conf.setAuthenticationEnabled(true);
+        conf.setAuthorizationEnabled(true);
+        conf.getProperties().setProperty("tokenSecretKey", "data:;base64,"
+                + Base64.getEncoder().encodeToString(SECRET_KEY.getEncoded()));
+        Set<String> superUserRoles = new HashSet<>();
+        superUserRoles.add("admin");
+        conf.setSuperUserRoles(superUserRoles);
+        Set<String> providers = new HashSet<>();
+        providers.add(AuthenticationProviderToken.class.getName());
+        conf.setAuthenticationProviders(providers);
+        super.internalSetup();
+        PulsarAdminBuilder pulsarAdminBuilder = PulsarAdmin.builder().serviceHttpUrl(brokerUrl != null
+                ? brokerUrl.toString() : brokerUrlTls.toString())
+                .authentication(AuthenticationToken.class.getName(),
+                        ADMIN_TOKEN);
+        admin = Mockito.spy(pulsarAdminBuilder.build());
+        admin.clusters().createCluster(testLocalCluster, new ClusterDataImpl());
+        admin.tenants().createTenant(testTenant, new TenantInfoImpl(Sets.newHashSet("role1", "role2"),
+                Sets.newHashSet(testLocalCluster)));
+        admin.namespaces().createNamespace(testTenant + "/" + testNamespace,
+                Sets.newHashSet(testLocalCluster));
+        admin.namespaces().grantPermissionOnNamespace(testTenant + "/" + testNamespace, "producer",
+                EnumSet.of(AuthAction.produce));
+        admin.namespaces().grantPermissionOnNamespace(testTenant + "/" + testNamespace, "consumer",
+                EnumSet.of(AuthAction.consume));
+    }
+
+    @Override
+    @AfterMethod
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    @DataProvider(name = "variations")
+    public static Object[][] variations() {
+        return new Object[][]{
+                {CONSUME_TOKEN, 401},
+                {PRODUCE_TOKEN, 200}
+        };
+    }
+
+    @Test(dataProvider = "variations")
+    public void testProduceToNonPartitionedTopic(String token, int status) throws Exception {
```

Comment:
There are some repeated code blocks, maybe we could try to reuse the same code block.